### PR TITLE
feat: abstract out confirmation dialog

### PR DESCRIPTION
**Pull request by vkbansal**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

**Review by devinder-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — DISMISSED** [review]
can you add a story demoing/documenting how to use this new hook?
and possible tests?

**Review by bhavya-sinha — APPROVED** [review]

File: src/components/ConfirmDialog/ConfirmationDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    customButtons,
+    ...rest
+  } = props
+
+  function closeWithFalse(): void {
+    onClose?.(false)
+  }
+
+  function closeWithTrue(): void {
+    onClose?.(false)
```
Comment (knagurski) [inline src/components/ConfirmDialog/ConfirmationDialog.tsx:63]: Is this not `closeWithTrue`?

File: src/hooks/useToggle.ts [hunk 2]
```diff
@@ -38,3 +38,30 @@ export function useToggleWithLocalStorage(localStorageKey: string, initialState
...
 }
+
+export interface UseToggle2Return {
+  isOpen: boolean
+  open(): void
+  close(): void
+  toggle(): void
+}
+
+export function useToggle2(init = false): UseToggle2Return {
```
Comment (knagurski) [inline src/hooks/useToggle.ts:49]: I **REALLY** hate naming like this. Can we figure out a better name?

**Review by knagurski — CHANGES_REQUESTED** [review]
Looks like there's a bug in the `closeWithTrue`

**Comment by knagurski** [comment]
> can you add a story demoing/documenting how to use this new hook?
> 
> and possible tests?

I agree here, if anything UI Core should have more stringent testing requirements than NGUI.

File: src/components/ConfirmDialog/ConfirmationDialog.tsx (hunk 1 above)
Comment (vkbansal) [reply src/components/ConfirmDialog/ConfirmationDialog.tsx]: Fixed

**Comment by vkbansal** [comment]
added story and test @knagurski @abhinavrastogi-harness

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by knagurski — APPROVED** [review]
Nice simple API 👍🏻

**Review by knagurski — APPROVED** [review]
Awesome work 🎉

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**